app/models/template.rb

```diff
@@ -80,6 +80,15 @@ def self.org_customizations(dmptemplate_id, org_id)
     Template.where(customization_of: dmptemplate_id, org_id: org_id).order(version: :desc).valid.first
   end
 
+  def self.get_latest_template_versions(orgs)
+    # get the unique dmptemplate_ids (template version families)
+    families = Template.valid.where(org: orgs, customization_of: nil).pluck(:dmptemplate_id)
+    # return the most current versions of each template version family
+    ids = []
+    families.map{ |fam| ids << Template.current(fam).id }
```

Comment:
Thanks for moving this but it still has scaleability concerns
The map used generates a large object in memory and then throws it out the object, never using it.
One query is run per template (template.current(fam)), which is an n+1 query issue.

I again propose this gets re-factored to the following code:

```id_query = 'select id from (select id, version , max(version) over (partition by dmptemplate_id) as v from templates order by dmptemplate_id) as foo where version = v;'
ids = []
Template.find_by_sql(id_query).each do |templ|
  ids << templ.id
end
Template.includes(:org).where(id: ids).order('orgs.name, templates.title')
```
This will run 2 queries and find all the templates instead of 2+n

Reply:
Thanks for the example @xsrust. That works for Postgres but MySQL doesn't seem to support OVER. I refactored a bit and this gets us the same desired result (one query to collect the unique dmptemplate_ids and the other to grab the current version for each dmptemplate_id)
```sql
def self.get_latest_template_versions(orgs)
    # get the unique dmptemplate_ids (template version families)
    families = Template.valid.where(org: orgs, customization_of: nil).pluck(:dmptemplate_id)

    # return the most current versions of each template version family
    id_query = id_query = "SELECT templates.dmptemplate_id, templates.id, current.version "\
               "FROM "\
               "(SELECT MAX(version) AS version, dmptemplate_id "\
                "FROM templates WHERE dmptemplate_id in (#{families.join(', ')}) "\
                "GROUP BY dmptemplate_id) as current "\
               "INNER JOIN templates ON current.version = templates.version AND current.dmptemplate_id = templates.dmptemplate_id;"

    ids = Template.find_by_sql(id_query).collect{|t| t.id }
    Template.includes(:org).where(id: ids).order('orgs.name, templates.title')
  end
```

Reply:
Ahh, thanks.  I didnt realise MySQL didnt support OVER.  
@jollopre suggested another alternative where the subqueries are re-factored into scopes so we can re-use each sub-query in other places, detailed in https://github.com/DMPRoadmap/roadmap/issues/928, but I'm happy to merge this